src/app/(user)/home/components/select-funding-wallet.tsx

```diff
@@ -0,0 +1,162 @@
+      acct.walletClientType !== 'privy' &&
+      acct.chainType === 'solana',
+  );
+
+  const linkedSolanaWalletEmbeddedWallet: EmbeddedWallet | undefined = linkedSolanaWallet ? {
+    id: linkedSolanaWallet.address,
+    name: 'Privy Linked Wallet',
+    publicKey: linkedSolanaWallet.address || '',
+    walletSource: 'PRIVY',
+    chain: 'SOLANA',
+    delegated: linkedSolanaWallet.delegated,
+    ownerId: user ? user.id : '',
+    active: false,
+  }: undefined;
+
+  const legacyWallets = embeddedWallets.filter(
+    (w: EmbeddedWallet) => w.walletSource === 'CUSTOM' && w.chain === 'SOLANA',
+  );
+
+  const allWalletAddresses = [
+    ...(linkedSolanaWallet ? [linkedSolanaWallet.address] : []),
+    ...legacyWallets.map((w) => w.publicKey),
+  ];
+
+  const [isLoading, setIsLoading] = useState(false);
+  const { fundWallet } = useFundWallet();
+
+  return (
+    <Dialog onOpenChange={onCancel} open={displayPrompt}>
+      <DialogContent className="w-full rounded-2xl p-4 shadow-xl sm:max-w-[650px] sm:p-6">
+        <DialogHeader>
+          <DialogTitle className="text-lg font-semibold sm:text-xl">
+            Select Funding Wallet
+          </DialogTitle>
+          <DialogDescription className="text-sm text-muted-foreground">
+            Choose how you want to fund this transaction
+          </DialogDescription>
+        </DialogHeader>
+
+        <div className="mt-4 space-y-6">
+          {/* Phantom Wallet Option */}
+          <div
+            className="cursor-pointer rounded-lg border border-border p-4 shadow-sm transition-colors hover:bg-muted sm:rounded-xl"
+            onClick={() => {
+              onSelectWallet('phantom');
+            }}
+          >
+            <div className="flex items-center gap-4">
+              <Image
+                src="/icons/phantom-icon.png"
+                alt="Phantom"
+                width={40}
+                height={40}
+                className="rounded-full"
+              />
+              <div className="space-y-0.5">
+                <h3 className="text-sm font-medium sm:text-base">
+                  Pay with Phantom
+                </h3>
+                <p className="text-xs text-muted-foreground sm:text-sm">
+                  Connect and pay using your Phantom wallet
+                </p>
+              </div>
+            </div>
+          </div>
+
+          {linkedSolanaWalletEmbeddedWallet && (
+            <div className="space-y-3">
+              <h3 className="text-xs font-semibold uppercase tracking-wide text-muted-foreground sm:text-sm">
+                Linked Solana Wallets
+              </h3>
+              <div className="space-y-2">
+                <WalletCardMini
+                  key={linkedSolanaWalletEmbeddedWallet.id}
+                  wallet={linkedSolanaWalletEmbeddedWallet}
+                  mutateWallets={mutateWallets}
+                  allWalletAddresses={allWalletAddresses}
+                  useWallet={() => {}}
```

Comment:
TODO: remaining step is to implement using the balance in the selected wallet for eap payments